target/cheri-common/op_helper_cheri_common.c

```diff
@@ -450,6 +450,7 @@ void CHERI_HELPER_IMPL(cbuildcap(CPUArchState *env, uint32_t cd, uint32_t cb,
         result.cr_perms = ctp->cr_perms;
         result.cr_uperms = ctp->cr_uperms;
         result._cr_cursor = ctp->_cr_cursor;
+        result.cr_ebt = ctp->cr_ebt;
```

Comment:
Missing flags and reserved still. But at this point why are we even using the authorising capability? Why not not just `cap_register_t result = *ctp; if (!cap_is_sealed_entry(ctp)) result.cr_otype = CAP_OTYPE_UNSEALED;` (though that's wrong on RISC-V, we don't want to unseal arbitrary sealed things, it only permits sentries and unsealed caps, always preserving the otype). There should be literally nothing inherited from the authorising capability other than architectural provenance.

Reply:
Glancing at the riscv sail, I don't see any conditioning on the `otype` beyond "is a sentry" (and if so, return a sentry, otherwise, return unsealed) (though it's missing the check for permit_execute when it does decide to return a sentry...).

Reply:
Sail is wrong; it has `  } else if cb_val.sealed then {  handle_cheri_reg_exception(CapEx_SealViolation, cb);` further up, and is also rederiving. Flute has the "right" implementation: https://github.com/CTSRD-CHERI/Flute/blob/2a8d8708c77e477e9be5d4938b4d825521de369f/src_Core/CPU/EX_ALU_functions.bsv#L1713-L1736

Reply:
Although I have now spotted a bug in the cap idx Flute reports for register 0... :)

Reply:
I believe both of the following statements, but both tentatively: `cb_val` is the authority, and it's correct to reject sealed authorities in `cbuildcap`?